src/dc_receive_imf.rs

```diff
@@ -1686,29 +1668,27 @@ unsafe fn check_verified_properties(
             // - OR if the verified-key does not match public-key or gossip-key
             //   (otherwise a verified key can _only_ be updated through QR scan which might be annoying,
             //   see https://github.com/nextleap-project/countermitm/issues/46 for a discussion about this point)
-            if 0 == is_verified
+            if !is_verified
                 || peerstate.verified_key_fingerprint != peerstate.public_key_fingerprint
                     && peerstate.verified_key_fingerprint != peerstate.gossip_key_fingerprint
             {
-                info!(context, "{} has verfied {}.", contact.get_addr(), to_addr,);
+                info!(context, "{} has verified {}.", contact.get_addr(), to_addr,);
                 let fp = peerstate.gossip_key_fingerprint.clone();
                 if let Some(fp) = fp {
                     peerstate.set_verified(0, &fp, 2);
-                    peerstate.save_to_db(&context.sql, false);
-                    is_verified = 1;
+                    peerstate.save_to_db(&context.sql, false).unwrap();
```

Comment:
i'd rather like dc_receive_imf return a result and bubble it up.  Carrying on sounds dubious, though -- we are missing a changed key which would break e2e-chatting afterwards.